pgraph/pgraph_test.go

```diff
@@ -49,6 +51,54 @@ func TestCount1(t *testing.T) {
 	}
 }
 
+func TestCountNodesAndEdges(t *testing.T) {
+
+	G := &Graph{Name: "MyG"}
+
+	if i := G.NumVertices(); i != 0 {
+		t.Errorf("should have 0 vertices instead of: %d", i)
+	}
+
+	if i := G.NumEdges(); i != 0 {
+		t.Errorf("should have 0 edges instead of: %d", i)
+	}
+
+	// populate vertices
+	var vertices [6]Vertex
+	for i := 0; i < len(vertices); i++ {
+		var name string = "v" + strconv.Itoa(i+1)
+		vertices[i] = NV(name)
+	}
+
+	// populate edges
+	var edges [5]Edge
+	for x := 0; x < len(edges); x++ {
+		var name string = "e" + strconv.Itoa(x+1)
+		edges[x] = NE(name)
+	}
+
+	// add all the vertices
+	for a := 0; a < len(vertices); a++ {
+		G.AddVertex(vertices[a])
+	}
+
+	// add edges to the vertices
+	for a := 0; a < len(vertices); a++ {
+		var b int = a + 1
+		if a <= len(vertices)-2 {
```

Comment:
Subtle: -2's don't usually appear in code as much as -1's. Since we're a zero-based language, it's expected to see a -1 here and there... But you're using the -2 off of the first of the two elements... So it raises an eyebrow... So probably best to use -1 and compare it to b instead.